haystack/nodes/base.py

```diff
@@ -74,6 +70,38 @@ def __init_subclass__(cls, **kwargs):
         # Enables generic load() for all specific component implementations.
         cls._subclasses[cls.__name__] = cls
 
+    @property
+    def name(self) -> Optional[str]:
+        if "name" not in self._component_config:
+            return None
+        return self._component_config["name"]
+
+    @name.setter
+    def name(self, value: str):
+        self._component_config["name"] = value
+
+    @property
+    def sub_components(self) -> List[BaseComponent]:
+        if "params" not in self._component_config:
+            return list()
+        return [param for param in self._component_config["params"].values() if isinstance(param, BaseComponent)]
+
+    @property
+    def type(self) -> str:
```

Comment:
Technically this property is redundant, `type(node)` would give you the same value is it?

I could not easily get `type` out of the `_component_config` dict because it required refactoring `get_config` deeply, but now that we're at it, it might be worth trying

Reply:
No, this type returns a string: so type(node).__name__ basically.
I also thought about moving name and type out of _component_config. But on the other hand we have the complete config in one place with it. I like that idea but I'm open for discussing this.